### PR TITLE
fix: remove incorrect optimization for group-by

### LINKED ISSUE #21639: Forward-port group by fix to 2.x
Forward port of https://github.com/influxdata/influxdb/issues/21630

### PR DESCRIPTION
**Pull request by lesam**
Closes #21639

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
@danxmoran is there a good place to add an end-to-end select test (would the grace tests be the right place for that?)

**Comment by danxmoran** [comment]
@lesam I'd vote for adding a new test to either `influxql/v1validation` or `influxql/v1tests`. The `v1validation` suite is data-driven based on `goldenfiles` there, so if you can express the input/query/output in YAML then it'll probably be the least amount of code overall.

You'd be the first person adding a test since Stuart ported the suite over from IDPE so you might hit some warts, but I think the overall framework is solid.

File: influxql/v1tests/query_test.go [hunk 1]
```diff
@@ -88,6 +89,45 @@ func TestServer_Query_Chunked(t *testing.T) {
 	test.Run(ctx, t, s)
 }
 
+// Ensure a more complex group-by is correct
+func TestServer_Query_ComplexGroupby(t *testing.T) {
```
Comment (lesam) [inline influxql/v1tests/query_test.go:93]: This test fails with the `fastIdx` 'optimisation'

**Review by danxmoran — APPROVED** [review]
Thanks!

**Pull request merged**